src/material/datepicker/datepicker-input.ts

```diff
@@ -6,364 +6,154 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {coerceBooleanProperty} from '@angular/cdk/coercion';
```

Comment:
I think maybe these changes are a merge conflict? Let's revert them and worry about integrating the `DateSelectionModel` with the `MatDatepickerInput` in a future PR